### PR TITLE
asTime fixed for time-only values

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13343

### CODE REVIEW

**Review by SilverFire — APPROVED** [review]

File: framework/i18n/Formatter.php [hunk 1]
```diff
@@ -650,40 +650,46 @@ private function formatDateTimeValue($value, $format, $type)
...
      * Defaults to `false`. If `true`, the method will then return an array with the first element being the normalized
-     * timestamp and the second a boolean indicating whether the timestamp has time information or it is just a date value.
+     * timestamp, the second a boolean indicating whether the timestamp has time information and third a boolean indicating 
+     * whether the timestamp has date information.
      * This parameter is available since version 2.0.1.
      * @return DateTime|array the normalized datetime value.
      * Since version 2.0.1 this may also return an array if `$checkTimeInfo` is true.
      * The first element of the array is the normalized timestamp and the second is a boolean indicating whether
      * the timestamp has time information or it is just a date value.
+     * Since version 2.0.12 the array has third boolean element indicating whether the timestamp has date information.
```
Comment (cebe) [inline framework/i18n/Formatter.php]: A note in the `UPGRADE` file is needed.

File: tests/framework/i18n/FormatterDateTest.php [hunk 2]
```diff
@@ -687,6 +687,18 @@ public function testDateOnlyValues()
...
+    public function testTimeOnlyValues()
+    {
+        $this->formatter->defaultTimeZone = 'UTC';
+        $this->formatter->timeZone = 'Europe/Berlin'; // UTC+1 (DST UTC+2)
+
+        // when timezone conversion is made on this time, it will result in 11:32 (DST 12:32) to be returned.
+        // ensure this does not happen on time only values
+        $this->assertSame('10:32', $this->formatter->asTime('10:32', 'HH:mm'));
+        
+        $this->assertSame('9am', $this->formatter->asTime('9am', 'Ha'));
```
Comment (cebe) [inline tests/framework/i18n/FormatterDateTest.php]: a test including seconds would be nice.

File: framework/i18n/Formatter.php (hunk 1 above)
Comment (bizley) [inline framework/i18n/Formatter.php]: `UPGRADE` draft added.

File: tests/framework/i18n/FormatterDateTest.php (hunk 2 above)
Comment (bizley) [inline tests/framework/i18n/FormatterDateTest.php]: Done.

**Review by SilverFire — APPROVED** [review]

**Review by dynasource — CHANGES_REQUESTED** [review]
could this PR be completed with a unit test which matches the clear use case of https://github.com/yiisoft/yii2/issues/13343#issuecomment-271119689 (including a reference?)

**Comment by SilverFire** [comment]
https://github.com/yiisoft/yii2/pull/13348/files#diff-f1eb247f183bf542c55a57d24506a469R698
Isn't this enough?

**Comment by bizley** [comment]
It's ok, I've just finished the exact use case test, commit in few seconds.

**Review by dynasource — APPROVED** [review]
thanks!

**Review by cebe — APPROVED** [review]

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**